completions/fio

```diff
@@ -24,13 +31,13 @@ _fio()
             return
             ;;
         --cmdhelp)
-            # TODO more commands?
-            COMPREPLY=($(compgen -W "all" -- "$cur"))
+            ret=($("$1" --cmdhelp=all 2>/dev/null | awk '{print $1}'))
+            COMPREPLY=($(compgen -W '${ret[@]} all' -- "$cur"))
             return
             ;;
         --enghelp)
-            # TODO print ioengine help, or list available ioengines
-            # TODO engine,help arg
+            _comp_cmd_fio__engines "$1"
+            COMPREPLY=($(compgen -W '${ret[@]}' -- "$cur"))
```

Comment:
```suggestion
            _comp_cmd_fio__engines "$1"
            ((${#ret[@]})) && COMPREPLY=($(compgen -W '"${ret[@]}"' -- "$cur"))
```

- Avoid redundant word splitting. Actually, I don't think we need to suppress word splitting in this particular case because the engine names are unlikely to include spaces, but I don't want to rely on this kind of an implicit assumption about the values of the variables.
- Avoid the `nounset` error in Bash 4.2 and 4.3 in case it fails. Another approach might be to let `_comp_cmd_fio__engines` return through its exit status whether it generated at least one engine name and
  ```suggestion
              _comp_cmd_fio__engines "$1" &&
                  COMPREPLY=($(compgen -W '"${ret[@]}"' -- "$cur"))
  ```

Reply:
Nice, I'll go with the latter approach for slightly less repeated code.